core/rfb.js

```diff
@@ -392,7 +392,10 @@ RFB.prototype = {
             uri = this._encrypt ? 'wss' : 'ws';
         }
 
-        uri += '://' + this._rfb_host + ':' + this._rfb_port + '/' + this._rfb_path;
+        var relpath = location.pathname.split('/');
```

Comment:
`core/rfb.js` *should not* depend on the current location -- that's something that should be handled in the UI.  If you want a relative path, do it in the UI code that sets the RFB path.

Reply:
So your vote is to reject this PR?

Reply:
yeah, if we do this, I'd rather see it done in `app/ui.js`.

Reply:
ui.js keeps the websocket path in localstorage and reload it from settings saved earlier. Just lookup for any other way to override the settings by url (e.g. https://mppsrc.ogcio.hksarg/twhtang/novnc/vnc.html?path=/context/novnc/websockify). It simplifies the integration for other web app to invoke novnc by opening new tab for different vnc console (e.g. window.open 'https://mppsrc.ogcio.hksarg/twhtang/novnc/vnc.html', '_blank').

Reply:
path overriding is already implemented in WebUitl.getQueryVar via query param. That is, no source code change is required. Thanks.